web/src/main/java/org/openmrs/web/Listener.java

```diff
@@ -258,28 +257,35 @@ public void contextInitialized(ServletContextEvent event) {
 			log.error(MarkerFactory.getMarker("FATAL"), "Failed to obtain JDBC connection", e);
 		}
 	}
-	
-	private void loadCsrfGuardProperties(ServletContext servletContext) throws FileNotFoundException, IOException {	
-		File file = new File(OpenmrsUtil.getApplicationDataDirectory(), "csrfguard.properties");
-		InputStream inputStream = null;
-		try {
-			inputStream = new FileInputStream(file);
-		}
-		catch (FileNotFoundException ex) {
-			final String fileName = servletContext.getRealPath("/WEB-INF/csrfguard.properties");
-			inputStream = new FileInputStream(fileName);
-			OutputStream outputStream = new FileOutputStream(file);
-			IOUtils.copy(inputStream, outputStream);
-		    IOUtils.closeQuietly(outputStream);
-		    IOUtils.closeQuietly(inputStream);
-		    
-		    //Moved to EOF by the copy operation. So open it again.
-		    inputStream = new FileInputStream(file);
-		}
-		Properties properties = new Properties();
-		properties.load(inputStream);
-		IOUtils.closeQuietly(inputStream);
-		CsrfGuard.load(properties);
+
+	private void loadCsrfGuardProperties(ServletContext servletContext) throws IOException {
+		File csrfGuardFile = new File(OpenmrsUtil.getApplicationDataDirectory(), "csrfguard.properties");
+		Properties csrfGuardProperties = new Properties();
+		if(csrfGuardFile.exists()) {
+			try(InputStream csrfGuardInputStream = Files.newInputStream(csrfGuardFile.toPath())) {
+				csrfGuardProperties.load(csrfGuardInputStream);
+			}
+			catch (Exception e) {
+				log.error(e.getMessage());
```

Comment:
I'm not sure about just logging an error here and continuing.  Why catch this Exception if you aren't going to fall back to other behavior?  If we want to log this here (which is probably a good idea), then I'd add some additional context in the logging message to make it easier for someone to know where the problem is and what went wrong, and then re-throw the Exception.